cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -693,6 +723,36 @@ func instanceStatusFromProvisioningState(provisioningState *string) *cloudprovid
 	return status
 }
 
+func powerStateFromStatuses(statuses []compute.InstanceViewStatus) string {
+	for _, status := range statuses {
+		if status.Code == nil || !isKnownPowerState(*status.Code) {
+			continue
+		}
+		return *status.Code
+	}
+
+	// PowerState is not set if the VM is still creating (or has failed creation),
+	// so the absence of a PowerState is treated the same as a VM that is stopped
+	return PowerStateStopped
+}
+
+func isRunningPowerState(powerState string) bool {
+	return powerState == PowerStateRunning || powerState == PowerStateStarting
+}
+
+func isKnownPowerState(powerState string) bool {
```

Comment:
I think functions like this and `isRunningPowerState()` could be in `azure_util.go` too. In this way we could manage the definitions of each power state from the same place.